Update workflow GitLab references to OAuth (#5345)
mintlify/docs: agent/workflows.mdx

For GitHub repositories, you must have the Mintlify GitHub App installed on every repository listed in the `context` or `on.push.repo` fields. Add new repositories on the [GitHub app](https://dashboard.mintlify.com/settings/organization/github-app) page of your Mintlify dashboard.

> For GitLab repositories, you must connect your GitLab account via OAuth in your organization's [GitLab settings](https://dashboard.mintlify.com/settings/organization/gitlab-oauth) and add the repositories you want to use. The user who connects GitLab OAuth must have at least the Maintainer role on each project.

Reviewer: ```suggestion
For GitLab repositories, you must connect your GitLab account on the [GitLab OAuth](https://dashboard.mintlify.com/settings/organization/gitlab-oauth) page in your Mintlify dashboard. Then add the repositories you want to use with workflows. You must have at least the Maintainer role on each project to connect GitLab OAuth .
```

<Note>
  For GitHub repositories, automerge requires the Mintlify GitHub App to have bypass permissions on all rulesets that target your deploy branch, including organization-level and repository-level rulesets. See [Configure automerge](/guides/configure-automerge) for setup instructions.

>   For GitLab repositories, automerge uses the GitLab OAuth connection configured in your organization's [GitLab settings](https://dashboard.mintlify.com/settings/organization/gitlab-oauth). The user who connected GitLab OAuth must have at least the Maintainer role on the project to merge merge requests.

Reviewer: ```suggestion
  For GitLab repositories, automerge uses the GitLab OAuth connection configured on the [GitLab OAuth](https://dashboard.mintlify.com/settings/organization/gitlab-oauth) page of your Mintlify dashboard. You must have at least the Maintainer role on each project to connect GitLab OAuth .
```
